package NCbbFast;

import "library/cpp/getoptpb/proto/confoption.proto";

message TConfig {
    option (NGetoptPb.Title) = "Cbb daemon";

    optional bool Debug = 1 [
        default = false,
        (NGetoptPb.Conf) = {
            Descr: "should run in debug mode",
            Short: 'v',
            Long: "verbose"
        }
    ];

    required string LogFile = 2 [
        (NGetoptPb.Conf) = {
            Descr: "Log file",
            Short: 'l',
            Long: "log"
        }
    ];

    required uint32 Port = 3 [
        (NGetoptPb.Conf) = {
            Descr: "port",
            Short: 'p',
            Long: "port"
        }
    ];

    optional uint32 Threads = 4 [
        default = 4,
        (NGetoptPb.Conf) = {
            Descr: "Num threads",
            Short: 't',
            Long: "num-threads"
        }
    ];

    repeated string DbHosts = 5 [
        (NGetoptPb.Conf) = {
            Descr: "Db hosts",
            Short: 's',
            Long: "db-hosts"
        }
    ];

    optional uint32 DbPort = 6 [
        default = 6432,
        (NGetoptPb.Conf) = {
            Descr: "Db port",
            Short: 'P',
            Long: "db-port"
        }
    ];

    optional string DbUser = 7 [
        (NGetoptPb.Conf) = {
            Descr: "Db User",
            Short: 'u',
            Long: "db-user"
        }
    ];

    optional string DbName = 8 [
        (NGetoptPb.Conf) = {
            Descr: "Db Name",
            Short: 'd',
            Long: "db-name"
        }
    ];

    optional string DbPassword = 9 [
        (NGetoptPb.Conf) = {
            Descr: "Db Password",
            Long: "db-password"
        }
    ];

    optional string DbSslMode = 10 [
        default = "verify-full",
        (NGetoptPb.Conf) = {
            Descr: "Db ssl mode",
            Long: "db-ssl-mode"
        }
    ];

    optional string TvmClientsList = 11 [
        default = "tvm_list",
        (NGetoptPb.Conf) = {
            Descr: "TVM client list",
            Long: "tvm-client-list"
        }
    ];

    optional string TvmClientCacheDir = 12 [
        default = "tvm_cache",
        (NGetoptPb.Conf) = {
            Descr: "TVM client cache dir",
            Long: "tvm-cache-dir"
        }
    ];

    optional uint32 TvmClientId = 13 [
        (NGetoptPb.Conf) = {
            Descr: "TVM client Id",
            Long: "tvm-client-id"
        }
    ];
}
